ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -79,178 +83,207 @@ interface State {
+                  fp.map('memory'),
+                  // See above comment on CPU union.
+                  fp.union([memory]),
+                  fp.sortBy(fp.identity)
+                )(validLeonardoMachineTypes)}
+                onChange={ 
+                  ({value}) => fp.flow(
+                    fp.find({cpu, memory: value}), 
+                    // If the selected machine is not different from the current machine return null
+                    maybeGetMachine,
+                    onChange
+                    )(validLeonardoMachineTypes) }
+                value={memory}
+                />
+    </div>
+  </Fragment>
+}
 
-    async componentDidMount() {
-      // TODO(RW-5420): Centralize a runtimeStore.
-      let runtime = null;
-      let error = false;
-      try {
-        const promise = runtimeApi().getRuntime(this.props.workspace.namespace, {signal: this.aborter.signal});
-        updateRuntimeOpsStoreForWorkspaceNamespace(this.props.workspace.namespace, {
-          promise: promise,
-          operation: 'get',
-          aborter: this.aborter
-        });
-        runtime = await promise;
-      } catch (e) {
-        // 404 is expected if the runtime doesn't exist, represent this as a null
-        // runtime rather than an error mode.
-        if (e.status !== 404) {
-          error = true;
-        }
-      }
-      markRuntimeOperationCompleteForWorkspace(this.props.workspace.namespace);
-      this.setState({
-        runtime,
-        error,
-        loading: false
-      });
-    }
+const DiskSizeSelection = ({onChange, selectedDiskSize, currentRuntime}) => {
+  const {dataprocConfig, gceConfig} = currentRuntime;
+  const masterDiskSize = !!dataprocConfig ? dataprocConfig.masterDiskSize : gceConfig.bootDiskSize
 
-    render() {
-      const {runtimeOps, workspace} = this.props;
-      const {loading, error, runtime} = this.state;
+  return <div>
+    <label htmlFor='runtime-disk'
+          style={{marginRight: '.25rem'}}>Disk (GB)</label>
+      <InputNumber id='runtime-disk'
+                //  disabled={true}
+                showButtons
+                decrementButtonClassName='p-button-secondary'
+                incrementButtonClassName='p-button-secondary'
+                value={selectedDiskSize || masterDiskSize}
+                inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+                onChange={({value}) => onChange(value === masterDiskSize ? null : value)}
+                min={50 /* Runtime API has a minimum 50GB requirement. */}/>
+  </div>
+}
+
+export const RuntimePanel = withCurrentWorkspace()(({workspace}) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [selectedDiskSize, setSelectedDiskSize] = useState(null);
+  const [selectedMachine, setselectedMachine] = useState(null);
+  const runtimeOps = useStore(runtimeOpsStore);
+  const [currentRuntime, setRequestedRuntime] = useRuntime(workspace.namespace);
```

Comment:
Usage of `useRuntime`.

Supply the `workspaceNamespace` and recieve the `currentRuntime` and a setter for a new runtime.
Any runtime updates that go through the underlying `runtimeStore` will trigger a render.

Ideally all runtime changes should go through this hook. Only the `LeoRuntimeInitializer` this hook, and this component (I am looking to factor that out) will directly reference the store.